hathor/builder.py

```diff
@@ -36,11 +40,266 @@
+        assert self._rocksdb_path is not None
+        if self._rocksdb_storage is None:
+            self._rocksdb_storage = RocksDBStorage(path=self._rocksdb_path)
+        return self._rocksdb_storage
+
+    def _create_tx_storage(self) -> TransactionStorage:
+        if self._tx_storage is not None:
+            return self._tx_storage
+        if self._storage_type == StorageType.MEMORY:
+            return TransactionMemoryStorage()
+        elif self._storage_type == StorageType.ROCKSDB:
+            rocksdb_storage = self._get_rocksdb_storage()
+            use_memory_index = self._force_memory_index
+            return TransactionRocksDBStorage(rocksdb_storage, use_memory_indexes=use_memory_index)
+        raise NotImplementedError
+
+    def use_memory(self) -> None:
+        self._storage_type = StorageType.MEMORY
+
+    def use_rocksdb(self, path: str) -> None:
+        self._storage_type = StorageType.ROCKSDB
+        self._rocksdb_path = path
+
+    def force_memory_index(self) -> None:
+        self._force_memory_index = True
+
+    def _create_wallet(self) -> Optional[BaseWallet]:
+        if self._wallet is not None:
+            assert self._wallet_directory is None
+            assert self._wallet_unlock is None
+            return self._wallet
+
+        if self._wallet_directory is None:
+            return None
+        wallet = Wallet(directory=self._wallet_directory)
+        if self._wallet_unlock is not None:
+            wallet.unlock(self._wallet_unlock)
+        return wallet
+
+    def set_wallet(self, wallet: BaseWallet) -> None:
+        self._wallet = wallet
+
+    def enable_keypair_wallet(self, directory: str, *, unlock: Optional[bytes] = None) -> None:
+        self._wallet_directory = directory
+        self._wallet_unlock = unlock
+
+    def enable_stratum_server(self, port: int) -> None:
+        self._stratum_port = port
+
+    def enable_address_index(self) -> None:
+        self._enable_address_index = True
+
+    def enable_tokens_index(self) -> None:
+        self._enable_tokens_index = True
+
+    def enable_utxo_index(self) -> None:
+        self._enable_utxo_index = True
+
+    def enable_wallet_index(self) -> None:
+        self.enable_address_index()
+        self.enable_tokens_index()
+
+    def set_tx_storage(self, tx_storage: TransactionStorage) -> None:
+        self._tx_storage = tx_storage
+
+    def set_reactor(self, reactor: Reactor) -> None:
+        self._reactor = reactor
+
+    def set_pubsub(self, pubsub: PubSubManager) -> None:
+        self._pubsub = pubsub
+
+    def set_network(self, network: str) -> None:
+        self._network = network
+
+    def enable_sync_v1(self) -> None:
+        self._enable_sync_v1 = True
+
+    def disable_sync_v1(self) -> None:
+        self._enable_sync_v1 = False
+
+    def enable_sync_v2(self) -> None:
+        self._enable_sync_v2 = True
+
+    def disable_sync_v2(self) -> None:
+        self._enable_sync_v2 = False
+
+
 class CliBuilder:
```

Comment:
Should we add a TODO to port `CliBuilder` to use the `Builder` interface? I think it could even be ported to a `Builder.from_args` classmethod (and the reactor could be a `set_*` method).

Reply:
Done!